server/lib/src/idm/server.rs

```diff
@@ -3589,6 +3658,137 @@ mod tests {
+                // Should not occur!
+                error!("A critical error has occurred! {:?}", e);
+                panic!();
+            }
+        };
+
+        idms_auth.commit().expect("Must not fail");
+
+        let mut idms_auth = idms.auth().await;
+        let anon_begin = AuthEvent::begin_mech(sid, AuthMech::Anonymous);
+
+        let r2 = idms_auth.auth(&anon_begin, ct, Source::Internal).await;
+
+        match r2 {
+            Ok(ar) => {
+                let AuthResult {
+                    sessionid: _,
+                    state,
+                } = ar;
+
+                match state {
+                    AuthState::Continue(allowed) => {
+                        // Check the uat.
+                        assert!(allowed.len() == 1);
+                        assert!(allowed.first() == Some(&AuthAllowed::Anonymous));
+                    }
+                    _ => {
+                        error!("A critical error has occurred! We have a non-continue result!");
+                        panic!();
+                    }
+                }
+            }
+            Err(e) => {
+                error!("A critical error has occurred! {:?}", e);
+                // Should not occur!
+                panic!();
+            }
+        };
+
+        idms_auth.commit().expect("Must not fail");
+
+        let mut idms_auth = idms.auth().await;
+        // Now send the anonymous request, given the session id.
+        let anon_step = AuthEvent::cred_step_anonymous(sid);
+
+        // Expect success
+        let r2 = idms_auth.auth(&anon_step, ct, Source::Internal).await;
+
+        let token = match r2 {
+            Ok(ar) => {
+                let AuthResult {
+                    sessionid: _,
+                    state,
+                } = ar;
+
+                match state {
+                    AuthState::Success(uat, AuthIssueSession::Token) => uat,
+                    _ => {
+                        error!("A critical error has occurred! We have a non-succcess result!");
+                        panic!();
+                    }
+                }
+            }
+            Err(e) => {
+                error!("A critical error has occurred! {:?}", e);
+                // Should not occur!
+                panic!();
+            }
+        };
+
+        idms_auth.commit().expect("Must not fail");
+
+        // Token_str to uat
+        // we have to do it this way because anonymous doesn't have an ideantity for which we cam get the expiry value
+        let Token::UserAuthToken(uat) = idms
+            .proxy_read()
+            .await
+            .validate_and_parse_token_to_token(Some(&token), ct)
+            .expect("Must not fail") else {
+            panic!("Unexpected auth token type for anonymous auth");
+            };
+
+        debug!(?uat);
+
+        assert!(
+            matches!(uat.expiry, Some(x) if dbg!(x) == OffsetDateTime::UNIX_EPOCH + ct + Duration::from_secs(new_authsession_expiry as u64))
```

Comment:
What does this dbg(x) here do?

Reply:
Add more pollution to your terminal when you run the test :p. It’s just a leftover from when I was fightings against the test, I’ll remove it